### PR TITLE
msg API changes

### PR DESCRIPTION
**Pull request by lukileczo**
## Description

Adapt to msg API changes
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/528
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: port/sockets.c [hunk 1]
```diff
@@ -530,120 +536,119 @@ static int socket_ioctl(int sock, unsigned long request, const void* in_data, vo
...
+		case SIOCGIFCONF: {
+			struct ifconf *ifconf = (struct ifconf *)out_data;
+			int maxlen = ifconf->ifc_len;
+			struct ifreq *ifreq = ifconf->ifc_req;
+			struct netif *netif;
 
-	case SIOCSIFTXQLEN:
-		return -EOPNOTSUPP;
+			ifconf->ifc_len = 0;
+			if (!ifreq)  // WARN: it is legal to pass NULL here (we should return the lenght sufficient for whole response)
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply port/sockets.c:586]: It's a valid comment

File: port/sockets.c [hunk 2]
```diff
@@ -1024,12 +1030,13 @@ static int do_getifaddrs(char *buf, size_t *buflen)
 	int i;
 #endif
 
-	NETIF_FOREACH(netif) {
+	NETIF_FOREACH(netif)
+	{
```
Comment (agkaminski) [inline port/sockets.c:1034]: Hmm, why this change? It adhered to our style

File: port/sockets.c [hunk 3]
```diff
@@ -996,7 +1001,8 @@ static int do_getaddrinfo(const char *name, const char *serv, const struct addri
 			memcpy(strdest, ai->ai_canonname, n);
 			dest->ai_canonname = (void *)(strdest - buf);
 			strdest += n;
-		} else
+		}
+		else
```
Comment (agkaminski) [inline port/sockets.c:1005]:
```suggestion
		else {
```

File: port/sockets.c [hunk 4]
```diff
@@ -1056,8 +1063,9 @@ static int do_getifaddrs(char *buf, size_t *buflen)
 
 	memset(buf, 0, needed);
 	memset(&sa, 0, sizeof(sa));
-	sin = (struct sockaddr_in *) &sa;
-	NETIF_FOREACH(netif) {
+	sin = (struct sockaddr_in *)&sa;
+	NETIF_FOREACH(netif)
+	{
```
Comment (agkaminski) [inline port/sockets.c:1068]:
```suggestion
	NETIF_FOREACH(netif) {
```

File: port/sockets.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukileczo) [reply port/sockets.c:586]: Oh, forgot about it

File: port/sockets.c (hunk 2 above)
Comment (lukileczo) [reply port/sockets.c:1034]: Apparently this is how our clang-format does it (I just ran it on this file), but I'll change it manually

Comment (nalajcie) [reply port/sockets.c]: foreach-alike macros need to be explicitly configured in clang-format (https://clang.llvm.org/docs/ClangFormatStyleOptions.html#foreachmacros) - then it would be formatted correctly by default :)
